accepted/2021/architecture-targeting.md

```diff
@@ -0,0 +1,496 @@
+
+- `dotnet build -r win-x64 -a arm64`
+- `dotnet build -r linux-arm64 --os win -a x64`
+
+The shorthand syntax is intended as a CLI and not MSBuild concept. For scenarios that today require specifying RIDs in project file, the intent is that users will continue to specify RIDs. If we find that there is a need for a shorthand RID syntax in project files, then we can consider extending this syntax to MSBuild.
+
+## Implementation timeline
+
+This proposal includes multiple breaking changes. They are impactful enough that we need to ensure a few key characteristics:
+
+- Breaks are introduced in an early preview of a new .NET version.
+- Breaks can be avoided by using existing gestures in a previous .NET version.
+- The breaks are configurable in some way.
+
+That leaves us with the following plan:
+
+- Add high-value additive changes for x64 emulation in .NET 6.
+- Add gestures that enable avoiding the .NET 7 breaking changes.
+- Make rest of changes, including all breaking changes, in .NET 7 Preview 1 or 2.
+- Publish plan before .NET 6 releases.
+
+The second point -- adding gestures that avoid future breaks -- is a key part of the plan. We also need to ensure that we update documentation and blogs to ensure that the users are aware of the changes that they should make when using .NET 6 in preparation for .NET 7 and later versions.
+
+### .NET 6
+
+The following changes should be included in .NET 6, motivated by the x64 emulation scenario. They are all additive and non-breaking.
+
+- Enable shorthand RID syntax. Add to the following verbs (priority order):
+   - `build`
+   - `publish`
+   - `tool install`
+   - `test`
+   - `run`
+   - `watch run`
+   - `watch test`
+- Do NOT enable the `-r` RID syntax on these same verbs. Doing so would cause a breaking change in .NET 7, which seems like a unforgivable mistake.
+- Enable implicit RID where there is one known choice:
+   - The macOS Arm64 SDK should treat pre .NET 6 versions as X64 by default.
+   - The Windows Arm64 SDK should treat pre .NET 5 versions as X64 by default.
+- Parity syntax between `dotnet build` and `dotnet publish` for pivoting between self-contained and framework-dependent modalities, for example:
+   - `dotnet build -r win-x64 --self-contained`
+   - `dotnet build -r win-x64 --self-contained true`
+   - `dotnet build -r win-x64 --self-contained false`
+   - `dotnet build -r win-x64 --no-self-contained`
+- Add warning when using `-r` without a `--self-contained` or `--no-self-contained`. The warning is for .NET 6+ apps only.
+
+The addition of the shorthand RID syntax and the parity syntax between `build` and `publish` will provide a satisfactory set of gestures to enable migration with .NET 6 to a non-breaking syntax with respect to .NET 7. In particular, users need to migrate any uses or `-r` to always be accompanied by one of the `--self-contained` or `--no-self-contained` switches.
```

Comment:
To confirm, if we add the shorthand to just build and publish, that would be sufficient for .net 6?  Sarah will start on this next week and I wanted to ensure we had the .net 6 part down.

Reply:
We need the shorthand syntax for all the verbs listed in the .NET 6 section. If not, we don't have a story for `test`, `run` and other experiences.

Reply:
Put another way, the spec is clear on ordering, including on that point. I'm happy to review any of those sections in more detail.